### ISSUE TITLE
If agent is offline have one-off commands and tasks be queued thru the "pending actions" mechanism

### ISSUE THREAD

**Issue by silversword411** [amidaware/tacticalrmm#1521]
If an agent is offline at the time of a command/script run is manually triggered it dies and disappears because the agent is offline

Have those items get put into the "pending actions" queue for when they come online

**Comment by RichardBurt65** [comment]
Scheduled scripts too would be good.

**Comment by asasin114** [comment]
Would love to see this implemented!

**Comment by Treawlony** [comment]
A must have for non-corporate environments (like schools) where laptops are not always online

**Comment by dfr-fands** [comment]
Seconding this, right now the most important feature in TacticalRMM we miss in our environment by far, as we have many sites in different timezones (we are paying the sponsor) and endpoints aren't always online. Would be great to have it in bulk scripts too (as per issue https://github.com/amidaware/tacticalrmm/issues/1060).

**Comment by LeoDJ** [comment]
I think I've found a workaround:

You can create a "run once" task via the policies, if you want to apply it to multiple agents. (And via the normal "Tasks" tab for a single agent, probably)
tl;dr: Create a policy, assign it to a site (if you don't have one already) and create a task with an "onboarding" schedule.

Step-by-step:
- Settings > Automation Manager
- Select (or create) policy
- Go to the "tasks" tab below
- Add Task
  - Give it a name
  - Add the action you want to perform (command / script, don't forget to press the add button at the end of the row)
  - Select "Onboarding" schedule
  - ->Add Task

This should now run the task after a few minutes on all online agents as well as offline agents after they're booted and checked in.
(At least from my preliminary tests. Will see if it works reliably in the next weeks.)

**Comment by dfr-fands** [comment]
> I think I've found a workaround (for Linux agents)

@LeoDJ I've been using this method for Windows as well, but it's extremely unreliable, sometimes it can take up to a day after the agents boot, other times the task does not run at all. I would say about a 50% success rate in getting the task to actually run when the computer boots. It's really not a viable alternative, for Windows at least.

**Comment by Kikk79** [comment]
This is the most annoying thing for me as well. On top of that, I think MeshCentral already has the capability built-in, and definitely with this plugin: https://github.com/ryanblenis/MeshCentral-ScriptTask
